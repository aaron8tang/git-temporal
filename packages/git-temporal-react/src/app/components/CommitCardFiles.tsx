import React from 'react';
import { style } from 'app/styles';

import { ICommitFile } from 'app/interfaces';

import { AddedDeleted } from 'app/components/AddedDeleted';

export interface CommitCardFilesProps {
  files: ICommitFile[];
  isExpanded?: boolean;
  style?: object;
}

const defaultContainerStyle = {
  marginBottom: 10,
};

const renderNone = () => {
  return <span>No files effected</span>;
};

const renderFile = (file: ICommitFile) => {
  return (
    <div style={{ marginBottom: 5, wordBreak: 'break-all' }}>
      <AddedDeleted
        linesAdded={file.linesAdded}
        linesDeleted={file.linesDeleted}
        style={{ marginLeft: 10 }}
      />
      <span>{file.name}</span>
    </div>
  );
};

const renderFiles = (files: ICommitFile[]) => {
  return files.map(renderFile);
};

const renderSummary = (files: ICommitFile[]) => {
  const noun = files.length === 1 ? 'file' : 'files';
  // this is rendered as a link to suggest that they can expand the card
  // and see all files changed.  The card itself handles the click though
  return (
    <span style={style('link')}>
      {files.length} {noun} changed
    </span>
  );
};

export const CommitCardFiles = (props: CommitCardFilesProps): JSX.Element => {
  return (
    <div style={style(defaultContainerStyle, props.style)}>
      {props.files.length <= 0
        ? renderNone()
        : props.isExpanded
          ? renderFiles(props.files)
          : renderSummary(props.files)}
    </div>
  );
};
