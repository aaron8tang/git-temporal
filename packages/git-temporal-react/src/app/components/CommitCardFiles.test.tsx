import React from 'react';
import { shallow } from 'enzyme';

import { CommitCardFiles } from './CommitCardFiles';

const testFiles = [
  {
    name: 'doc/api/crypto.md',
    linesAdded: 1,
    linesDeleted: 1,
  },
  {
    name: 'src/node_crypto.cc',
    linesAdded: 2,
    linesDeleted: 5,
  },
  {
    name: 'test/parallel/test-crypto-authenticated.js',
    linesAdded: 26,
    linesDeleted: 0,
  },
];

describe('components/CommitCardFiles', () => {
  describe('when rendered with files and expanded=false', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <CommitCardFiles files={testFiles} isExpanded={false} />
      );
    });

    test('it should match snapshot (it should have rendered "3 files changed")', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with files and expanded=true', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <CommitCardFiles files={testFiles} isExpanded={true} />
      );
    });

    test('it should match snapshot (it should have rendered 3 files with added and deleted)")', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered without files', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<CommitCardFiles files={[]} isExpanded={true} />);
    });

    test('it should match snapshot (it should have rendered a placeholder message)")', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
