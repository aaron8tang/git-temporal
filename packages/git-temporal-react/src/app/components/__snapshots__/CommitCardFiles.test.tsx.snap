// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CommitCardFiles when rendered with files and expanded=false it should match snapshot (it should have rendered "3 files changed") 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <span
    style={
      Object {
        "color": "blue",
        "cursor": "pointer",
        "textDecoration": "underline",
      }
    }
  >
    3
     
    files
     changed
  </span>
</div>
`;

exports[`components/CommitCardFiles when rendered with files and expanded=true it should match snapshot (it should have rendered 3 files with added and deleted)") 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <div
    key="0"
    style={
      Object {
        "wordBreak": "break-all",
      }
    }
  >
    <AddedDeleted
      linesAdded={1}
      linesDeleted={1}
    />
    <EllipsizedFileName
      fileName="doc/api/crypto.md"
      maxCharacters={58}
      style={
        Object {
          "fontSize": 11,
          "fontWeight": "normal",
          "lineHeight": 1.35,
        }
      }
    />
  </div>
  <div
    key="1"
    style={
      Object {
        "wordBreak": "break-all",
      }
    }
  >
    <AddedDeleted
      linesAdded={2}
      linesDeleted={5}
    />
    <EllipsizedFileName
      fileName="src/node_crypto.cc"
      maxCharacters={58}
      style={
        Object {
          "fontSize": 11,
          "fontWeight": "normal",
          "lineHeight": 1.35,
        }
      }
    />
  </div>
  <div
    key="2"
    style={
      Object {
        "wordBreak": "break-all",
      }
    }
  >
    <AddedDeleted
      linesAdded={26}
      linesDeleted={0}
    />
    <EllipsizedFileName
      fileName="test/parallel/test-crypto-authenticated.js"
      maxCharacters={58}
      style={
        Object {
          "fontSize": 11,
          "fontWeight": "normal",
          "lineHeight": 1.35,
        }
      }
    />
  </div>
</div>
`;

exports[`components/CommitCardFiles when rendered without files it should match snapshot (it should have rendered a placeholder message)") 1`] = `
<div
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <span>
    No files effected
  </span>
</div>
`;
